### PR TITLE
Add Abap Doc annotations to types

### PR DESCRIPTION
**Pull request by huber-nicolas**

### CODE REVIEW

**Comment by schneidermic0** [comment]
@huber-nicolas Would you also add the corresponding JSON schema file, please?

**Comment by huber-nicolas** [comment]
> @huber-nicolas Would you also add the corresponding JSON schema file, please?

done. See the newest commit

File: file-formats/intf/if_aff_types_v1.intf.abap [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      "! <p class="shorttext">ABAP Language Version (source code object)</p>
+      "! ABAP language version for source code objects
+      "! $required
+      abap_language_version TYPE ty_abap_language_version_sco,
+    END OF ty_header_60_sco.
+
+  TYPES:
+    "! <p class="shorttext">Header for Non-Source Code Objects</p>
+    "! The header for an ABAP main object (without source code) with a description of 60 characters
+    BEGIN OF ty_header_60,
```
Comment (schneidermic0) [inline file-formats/intf/if_aff_types_v1.intf.abap:76]: Should we remove this header, until it is needed by any type?

File: file-formats/intf/if_oo_aff_intf_v1.intf.abap [hunk 2]
```diff
@@ -0,0 +1,130 @@
+INTERFACE if_oo_aff_intf_v1 PUBLIC.
```
Comment (schneidermic0) [inline file-formats/intf/if_oo_aff_intf_v1.intf.abap:1]: The interface shall start with `Z`

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
Thanks, see my comments

File: file-formats/intf/intf.json [hunk 3]
```diff
@@ -195,46 +195,46 @@
...
           },
           "parameters": {
-            "type": "array",
             "title": "Parameter Descriptions",
-            "description": "Parameter descriptions of the method",
+            "description": "Parameter descriptions",
+            "type": "array",
             "items": {
+              "title": "Component Description",
+              "description": "Component description (e.g., for methods, attributes, types)",
```
Comment (schneidermic0) [inline file-formats/intf/intf.json:224]: Here, we were more specific before. Before the title was `Parameter`, now it is `Component Description`.

File: file-formats/intf/if_oo_aff_intf_v1.intf.abap (hunk 2 above)
Comment (wurzka) [reply file-formats/intf/if_oo_aff_intf_v1.intf.abap]: We have changed this

File: file-formats/intf/if_aff_types_v1.intf.abap (hunk 1 above)
Comment (wurzka) [reply file-formats/intf/if_aff_types_v1.intf.abap]: I added the ##needed pragma

Comment (schneidermic0) [reply file-formats/intf/if_aff_types_v1.intf.abap]: Ok, we should remove the pragma as soon as it is consumed by other object types

Comment (larshp) [reply file-formats/intf/if_aff_types_v1.intf.abap]: IMHO, dont add stuff to `main` that is not used, keep it in a draft PR, or copy the code to an issue

Comment (larshp) [reply file-formats/intf/if_aff_types_v1.intf.abap]: I'd even go as far as adding the NEEDED pragma to https://rules.abaplint.org/forbidden_pseudo_and_pragma/ in the configuration

**Comment by wurzka** [comment]
@schneidermic0 you can start your re-review now. There is only the Problem with abaplint which claims that I do not use two types of zif_aff_types_v1 (although they are used in zif_oo_aff_intf_v1 )

**Comment by larshp** [comment]
false positives fixed, only the first file was found by the [glob](https://www.npmjs.com/package/glob) statement, https://github.com/SAP/abap-file-formats/pull/99/commits/ae26098fad103b66bfb3b270fb486ed5932d847a

File: file-formats/intf/if_aff_types_v1.intf.abap [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  CONSTANTS:
+    "! <p class="shorttext">ABAP Language Version (Non-Source Code Objects)</p>
+    "! ABAP language version for non-source code objects
+    BEGIN OF co_abap_language_version,
+      "! <p class="shorttext">Standard</p>
+      "! Standard
+      standard          TYPE ty_abap_language_version VALUE ' ',
+      "! <p class="shorttext">Key User</p>
+      "! Key user
```
[the remark this answers is not in the document]
Comment (schneidermic0) [reply file-formats/intf/if_aff_types_v1.intf.abap]: This isn't fixed, yet. Is it?

File: file-formats/typesUsedForAll/zif_aff_types_v1.intf.abap [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  CONSTANTS:
+    "! <p class="shorttext">ABAP Language Version (Source Code Objects)</p>
+    "! "! ABAP language version for source code objects. For non-source-code objects use
+    "! {@link data: ty_abap_language_version}
+    BEGIN OF co_abap_language_version_src,
+      "! <p class="shorttext">Standard</p>
+      "! Standard
+      standard          TYPE ty_abap_language_version_src VALUE 'X',
+      "! <p class="shorttext">Key User</p>
+      "! Key user
```
Comment (schneidermic0) [inline file-formats/typesUsedForAll/zif_aff_types_v1.intf.abap:17]: I would still change this
```suggestion
      "! <p class="shorttext">ABAP for Key Users</p>
      "! ABAP for key user extensibility
```

Comment (wurzka) [reply file-formats/typesUsedForAll/zif_aff_types_v1.intf.abap]: Done.

**Review by schneidermic0 — APPROVED** [review]
LGTM, Thanks :)

**Pull request merged**